### PR TITLE
feat(pre-loader): hook into pre-loader within guard

### PR DESCRIPTION
**Pull request by christianmemije**
- Hooks into pre-loader within guard to also show pre-loader while authentication is being checked.
- Address https://github.com/Teradata/vantage-ui-platform/pull/42/files#r373169497
- Fixes https://github.com/Teradata/vantage-ui-platform/issues/39
- BREAKING CHANGE: Removes IUser from Session service that was duplicated in the User service.

### CODE REVIEW

File: src/lib/auth/guards/authentication.guard.ts [hunk 1]
```diff
@@ -13,21 +12,43 @@ export class VantageAuthenticationGuard implements CanActivate {
...
+      }),
+      map(() => true),
+      finalize(() => hidePreLoader()),
+    );
+  }
+}
+
+// PRE_LOADER_COUNT variable is defined inside pre-loader.html
+
+function showPreLoader(): void {
```
Comment (christianmemije) [inline src/lib/auth/guards/authentication.guard.ts]: I'm thinking we should move these functions into some other place...

Comment (emoralesb05) [reply src/lib/auth/guards/authentication.guard.ts]: Agreed, thoughts on where?

Comment (christianmemije) [reply src/lib/auth/guards/authentication.guard.ts]: Ideally outside auth package, but ng-packagr build issues incoming in 5 4 3 2....

Comment (christianmemije) [reply src/lib/auth/guards/authentication.guard.ts]: Well if we figure out https://github.com/Teradata/vantage-ui-platform/issues/39, then this should be possible too.

Comment (christianmemije) [reply src/lib/auth/guards/authentication.guard.ts]: I was thinking a pre-loader package, but we want this guard to still work even if the pre-loader is not used?

Comment (emoralesb05) [reply src/lib/auth/guards/authentication.guard.ts]: Even if we call the functions, wouldnt the guard just work either way?

File: src/lib/utilities/pre-loader/pre-loader.html [hunk 2]
```diff
@@ -15,12 +15,16 @@
     ~1 sec on dev mode in simple app
     So instead we look for ng-version to verify that the app is ready
     */
+    // global variable that can be accessed elsewhere in angular app
+    var PRE_LOADER_COUNT = 1;
```
[the remark this answers is not in the document]
Comment (christianmemije) [reply src/lib/utilities/pre-loader/pre-loader.html]: Good point, will prefix variable.

File: src/lib/auth/guards/authentication.guard.ts (hunk 1 above)
Comment (christianmemije) [reply src/lib/auth/guards/authentication.guard.ts]: Moved into pre-loader package.

File: src/lib/utilities/pre-loader/pre-loader.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (christianmemije) [reply src/lib/utilities/pre-loader/pre-loader.html]: Changed.

**Comment by owilliams320** [comment]
LGTM

**Pull request merged**